logicrunner/requestsfetcher.go

```diff
@@ -120,9 +120,16 @@ func (rf *requestsFetcher) fetch(ctx context.Context) error {
 		default:
 		}
 
-		requestCtx := freshContextFromContextAndRequest(ctx, *request)
-		tr := NewTranscript(requestCtx, reqRef, *request)
-		rf.broker.AddRequestsFromLedger(ctx, tr)
+		switch {
+		case incoming != nil:
```

Comment:
seems like you wanna use interface and type casting instead of two variables and nil check

Reply:
ack